util/src/main/java/io/camunda/zeebe/util/FatalErrorHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import org.slf4j.Logger;
+
+public final class FatalErrorHandler implements UncaughtExceptionHandler {
+  private final Logger log;
+
+  public FatalErrorHandler(final Logger log) {
+    this.log = log;
+  }
+
+  public void handleError(final Throwable e) {
+    if (e instanceof VirtualMachineError) {
+      log.error("An unrecoverable VM error was thrown, exiting now.", e);
```

Comment:
:wrench: Operators are unlikely to read our Javadoc, so we might want to expand the error message to explain why this class of error is unrecoverable and why we decide to exit. If we had a public troubleshooting guide we'd probably want to link it here as well, but I guess there's no user facing doc to link here :disappointed: I'm open to any ideas or suggestions you have here though. But I'm just thinking, if I'm an SRE and I'm deploying a Zeebe cluster, and I run into this, do I know what I have to do?